### PR TITLE
feat: add opentelemetry filter

### PR DESCRIPTION
**Pull request by h1z3y3**
Add opentelemetry filter

### CODE REVIEW

**Comment by h1z3y3** [comment]
@flycash  Could you please help me to review this PR?

**Comment by flycash** [comment]
sorry, I am busy these days. I will review it!
and please help to fix the conflicts

**Comment by flycash** [comment]
Looks good! And can you add a change log to CHANGELOG file?

**Comment by h1z3y3** [comment]
Shall I use 2.0.2-beta1 or new version 2.0.3 ?

**Comment by flycash** [comment]
Hi, please check golangci-lint issues, you can find them by click `Files changed` tab of this page.
And then, **do remember** that add a change log to CHANGELOG file

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # developing
 - [upgrade redisgo to v1.8.8](https://github.com/beego/beego/pull/4872)
 - [fix prometheus CVE-2022-21698](https://github.com/beego/beego/pull/4878)
+
+# v2.0.3
```
[the remark this answers is not in the document]
Comment (h1z3y3) [reply CHANGELOG.md]: I notice that only go1.14 can't pass the test.
So why does beego still need so such an old go version?

**Comment by flycash** [comment]
we got error `client/httplib/filter/opentelemetry/filter_test.go:36:16: undefined: io.NopCloser` for Go 14, Go 15, I am afraid Beego cannot use `io.NopCloser`

**Comment by h1z3y3** [comment]
> we got error `client/httplib/filter/opentelemetry/filter_test.go:36:16: undefined: io.NopCloser` for Go 14, Go 15, I am afraid Beego cannot use `io.NopCloser`

I replace it by `http.NoBody`

**Comment by flycash** [comment]
`mod/github.com/go-logr/logr@v1.2.3/funcr/funcr.go:398:16: undefined: strconv.FormatComplex` it seems that Go1.14 do not have this function

**Comment by h1z3y3** [comment]
> `mod/github.com/go-logr/logr@v1.2.3/funcr/funcr.go:398:16: undefined: strconv.FormatComplex` it seems that Go1.14 do not have this function

https://github.com/go-logr/logr/issues/139

Why we still support so such old go version ?

**Comment by flycash** [comment]
> > `mod/github.com/go-logr/logr@v1.2.3/funcr/funcr.go:398:16: undefined: strconv.FormatComplex` it seems that Go1.14 do not have this function
> 
> I ask here: [go-logr/logr#139](https://github.com/go-logr/logr/issues/139)
> 
> Why we still support so such old go version ?

I think this is a good question and I just bump up the version to 1.18, please to rebase the master

**Pull request merged**